libs/qCC_db/src/ccPointCloud.cpp

```diff
@@ -1801,6 +1801,242 @@ bool ccPointCloud::colorize(float r, float g, float b, float a/*=1.0f*/)
 	return true;
 }
 
+//! "Cellular" function to apply a gaussian filter on the RGB values of points inside an octree cell
+/** This function is meant to be applied to all cells of the octree
+	The method also permits to use a bilateral behaviour for the filter. This is automatically switched on
+	if its sigmaSF parameter in additionalParameters is positive.
+
+	See ccPointCloud::applyScalarFieldGaussianFilter.
+	Method parameters (defined in "additionalParameters") are :
+	- (PointCoordinateType*) sigma
+	- (PointCoordinateType*) sigmaSF
+
+	\param cell structure describing the cell on which processing is applied
+	\param additionalParameters see method description
+	\param nProgress optional (normalized) progress notification (per-point)
+**/
+static bool ComputeCellGaussianFilter(	const CCCoreLib::DgmOctree::octreeCell& cell,
+										void** additionalParameters,
+										CCCoreLib::NormalizedProgress* nProgress = nullptr )
+{
+	//additional parameters
+	PointCoordinateType sigma = *(static_cast<PointCoordinateType*>(additionalParameters[0]));
+	PointCoordinateType sigmaSF = *(static_cast<PointCoordinateType*>(additionalParameters[1]));
+	bool applyToSF = *(static_cast<bool*>(additionalParameters[2]));
+	unsigned char burntOutColorThresholdMin = *(static_cast<unsigned char*>(additionalParameters[3]));
+	unsigned char burntOutColorThresholdMax = 255 - burntOutColorThresholdMin;
+	bool median = *(static_cast<bool*>(additionalParameters[4]));
```

Comment:
Is it really a 'median' filter? Not a simple average?

Reply:
You are right, it is average not median.

However, real median maybe even better :D but that is for another time i think :)